pennylane_qiskit/qiskit_device2.py

```diff
@@ -0,0 +1,484 @@
+}
+
+def accepted_sample_measurement(m: qml.measurements.MeasurementProcess) -> bool:
+    """Specifies whether or not a measurement is accepted when sampling."""
+    return isinstance(
+        m,
+        (
+            qml.measurements.SampleMeasurement,
+            qml.measurements.ClassicalShadowMP,
+            qml.measurements.ShadowExpvalMP,
+        ),
+    )
+
+@transform
+def split_measurement_types(
+    tape: qml.tape.QuantumTape,
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """Split into seperate tapes based on measurement type. Counts will use the
+    Qiskit Sampler, ExpectationValue and Variance will use the Estimator, and other
+    strictly sample-based measurements will use the standard backend.run function"""
+
+    use_sampler = [mp for mp in tape.measurements if isinstance(mp, CountsMP)]
+    use_estimator = [mp for mp in tape.measurements if isinstance(mp, (ExpectationMP, VarianceMP))]
+    other = [mp for mp in tape.measurements if not isinstance(mp, (CountsMP, ExpectationMP, VarianceMP))]
+
+    output_tapes = []
+    if use_sampler:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, use_sampler, shots=tape.shots))
+    if use_estimator:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, use_estimator, shots=tape.shots))
+    if other:
+        output_tapes.append(qml.tape.QuantumScript(tape.operations, other, shots=tape.shots))
+
+    return tuple(output_tapes), null_postprocessing
+
+@transform
+def validate_measurement_types(
+        tape: qml.tape.QuantumTape,
+    ) -> (Sequence[qml.tape.QuantumTape], Callable):
+
+    measurement_types = set(type(mp) for mp in tape.measurements)
+
+    if measurement_types.issubset({ExpectationMP, VarianceMP}):
+        return (tape,), null_postprocessing
+    elif measurement_types.issubset({CountsMP}):
+        return (tape,), null_postprocessing
+    else:
+        if measurement_types.intersection({CountsMP, ExpectationMP, VarianceMP}):
+            raise RuntimeError("Bad measurement combination")
+
+    return (tape,), null_postprocessing
+
+
+
+
+class QiskitDevice2(Device):
+    r"""Abstract Qiskit device for PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``).
+        backend (str | Backend): the initialized Qiskit backend
+
+    Keyword Args:
+        shots (int or None): number of circuit evaluations/random samples used
+            to estimate expectation values and variances of observables.
+        use_primitives(bool): whether or not to use Qiskit Primitives and the Qiskit Runtime Session. Defaults to False.
+    """
+    name = "Qiskit PennyLane plugin"
+    pennylane_requires = ">=0.30.0"
+    version = __version__
+    plugin_version = __version__
+    author = "Xanadu"
```

Comment:
are all of these needed with the new dev API?

Reply:
I believe no